x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -200,16 +201,15 @@ private void runJob(JobTask jobTask, JobState jobState, OpenJobAction.JobParams
             return;
         }
 
-        ActionListener<DatafeedContext> datafeedContextListener = ActionListener.wrap(
-            datafeedContext -> {
-                if (datafeedContext != null && datafeedContext.shouldRecoverFromCurrentSnapshot()) {
-                    // This job has a datafeed attached to it and the job had advanced past the current model snapshot.
+        ActionListener<Boolean> hasRunningDatafeedTaskListener = ActionListener.wrap(
+            hasRunningDatafeed -> {
+                if (hasRunningDatafeed) {
+                    // This job has a running datafeed attached to it.
                     // In order to prevent gaps in the model we revert to the current snapshot deleting intervening results.
-                    ModelSnapshot modelSnapshot = datafeedContext.getModelSnapshot() == null ?
-                        ModelSnapshot.emptySnapshot(jobTask.getJobId()) : datafeedContext.getModelSnapshot();
-                    logger.info("[{}] job had advanced past its current model snapshot [{}]; performing recovery",
-                        jobTask.getJobId(), modelSnapshot.getSnapshotId());
-                    revertToSnapshot(modelSnapshot, ActionListener.wrap(
+                    String modelSnapshotId = params.getJob().getModelSnapshotId() == null ?
+                        ModelSnapshot.EMPTY_SNAPSHOT_ID : params.getJob().getModelSnapshotId();
+                    logger.info("[{}] job has running datafeed task; performing recovery", jobTask.getJobId());
```

Comment:
Can you change "performing recovery" to something like "reverting to last good model snapshot [X]"

But only if `modelSnapshotId != ModelSnapshot.EMPTY_SNAPSHOT_ID`

Reply:
Right after that log message, we'll get the logging from the revert action which follows up and explains the recovery. I thought that was nice. Does it make sense or you still prefer to mention we're going to revert in there?

Reply:
OK, changed it to "reverting to current snapshot"